### PR TITLE
Updated customer addbank

### PR DESCRIPTION
**Pull request by ankitdas13**
Doc reference

[Add Bank Account](https://razorpay.com/docs/api/customers/bank-accounts/#2-delete-bank-account-of-customer)
[Delete Bank Account](https://razorpay.com/docs/api/customers/bank-accounts/#2-delete-bank-account-of-customer)
[Eligibility API - Affordability](https://razorpay.com/docs/payments/payment-gateway/affordability/eligibility-check/#eligibility-check-api)
[Fetch Eligibility- Affordability](https://razorpay.com/docs/payments/payment-gateway/affordability/eligibility-check/#fetch-eligibility-by-id)

### CODE REVIEW

File: lib/razorpay/customer.rb [hunk 1]
```diff
@@ -38,5 +38,21 @@ def fetchToken(tokenId)
     def deleteToken(tokenId)
       self.class.request.delete "#{id}/tokens/#{tokenId}" 
     end
+
+    def addBankAccount(options = {})
```
Comment (bhavyay) [inline lib/razorpay/customer.rb:42]: general practice in ruby is to use snake case. In this sdk as well, we have mostly used snake case. Can we follow the same naming convention here as well ?

File: test/razorpay/test_customer.rb [hunk 2]
```diff
@@ -65,5 +67,56 @@ def test_customer_delete_token
       token = Razorpay::Customer.fetch(@customer_id).deleteToken("token_FHfn3rIiM1Z8nr")
       assert token.deleted
     end
+
+    def test_customer_add_bank_account
```
Comment (bhavyay) [inline test/razorpay/test_customer.rb:71]: please add test cases for error scenarios as well for each of the APIs added.

File: documents/customer.md [hunk 3]
```diff
@@ -157,6 +157,249 @@ Razorpay::Customer.fetch(customerId)
...
+| beneficiary_mobile | integer      | Mobile number of the beneficiary.  |
+| beneficiary_city  | string      | The name of the city of the beneficiary.  |
+| beneficiary_state | string      | The state of the beneficiary.  |
+| beneficiary_country | string      | The country of the beneficiary.  |
+| beneficiary_pin   | interger    | The pin code of the beneficiary's address.  |
+| ifsc_code          | string      | The IFSC code of the bank branch associated with the account.  |
+
+**Response:**
+```json
+{
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/customer.md:199]: response updated

File: lib/razorpay/customer.rb (hunk 1 above)
Comment (ankitdas13) [reply lib/razorpay/customer.rb]: updated functions to snake case and removed chaining

File: test/razorpay/test_customer.rb (hunk 2 above)
Comment (ankitdas13) [reply test/razorpay/test_customer.rb:71]: added error test cases

File: documents/customer.md [hunk 4]
```diff
@@ -157,6 +157,249 @@ Razorpay::Customer.fetch(customerId)
...
+    "beneficiary_mobile"    : "8762489310",
+    "beneficiary_city"      :"Bangalore",
+    "beneficiary_state"     : "KA",
+    "beneficiary_country"   : "IN"
+})
+```
+
+**Parameters:**
+
+| Name          | Type        | Description                                 |
```
[the remark this answers is not in the document]
Comment (bhavyay) [reply documents/customer.md:183]: can you format this table as well ?

File: documents/customer.md [hunk 5]
```diff
@@ -157,6 +157,248 @@ Razorpay::Customer.fetch(customerId)
...
+}
+```
+
+-------------------------------------------------------------------------------------------------------
+
+### Fetch Eligibility by id
+
+```rb
+eligibilityId = "elig_F1cxDoHWD4fkQt"
+Razorpay::Customer.fetchEligibility(eligibilityId)
```
Comment (bhavyay) [inline documents/customer.md:360]: please update the function name here

File: documents/customer.md [hunk 6]
```diff
@@ -157,6 +157,248 @@ Razorpay::Customer.fetch(customerId)
...
+    "status": "deleted"
+}
+```
+
+-------------------------------------------------------------------------------------------------------
+
+### Eligibility Check API
+
+```rb
+Razorpay::Customer.requestEligibilityCheck({
```
Comment (bhavyay) [inline documents/customer.md:246]: please update the function name here

Comment (ankitdas13) [reply documents/customer.md]: updated

https://github.com/razorpay/razorpay-ruby/blob/981d25ccff2474c3b1cbed16657d6a45606f7d35/documents/customer.md?plain=1#L246

File: documents/customer.md (hunk 5 above)
Comment (ankitdas13) [reply documents/customer.md]: updated

https://github.com/razorpay/razorpay-ruby/blob/981d25ccff2474c3b1cbed16657d6a45606f7d35/documents/customer.md?plain=1#L360

**Review by bhavyay — APPROVED** [review]

**Pull request merged**